service/geopmdpy/session.py

```diff
@@ -31,91 +31,563 @@
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 
+"""Implementation for the geopmsession command line tool
+
+"""
+
 import sys
+import os
 import time
-from dasbus.connection import SystemMessageBus
+import math
 from argparse import ArgumentParser
+from dasbus.connection import SystemMessageBus
+from dasbus.error import DBusError
 from . import topo
+from . import pio
+
+
+class Session:
+    """Object responsible for creating a GEOPM service session
+
+    This object's run() method is the main entry point for
+    geopmsession command line tool.  The inputs to run() are derived
+    from the command line options provided by the user.
+
+    The Session object depends on the Requests object to parse the
+    input request buffer from the user.  The Session object also
+    depends on the TimedLoop object when executing a periodic read
+    session.
+
+    """
+
+    def __init__(self, geopm_proxy):
+        """Constructor for Session class
+
+        Args:
+            geopm_proxy (dasbus.client.proxy.InterfaceProxy): The
+                dasbus proxy for the GEOPM D-Bus interface.
+
+        Raises:
+            RuntimeError: The geopm systemd service is not running
+
+        """
+        try:
+            geopm_proxy.PlatformOpenSession
+        except DBusError as ee:
+            if 'io.github.geopm was not provided' in str(ee):
+                err_msg = """The geopm systemd service is not enabled.
+    Install geopm service and run 'systemctl start geopm'"""
```

Comment:
```suggestion
                err_msg = """The geopm systemd service is not enabled.
                             Install geopm service and run 'systemctl start geopm'"""
```

Reply:
We don't want that much white space in the printed message, I thought a 4 space indent looks okay.

Reply:
I really don't like the indentation.  Do we have this pattern any where else?

What about this instead?
```suggestion
                err_msg = "The geopm systemd service is not enabled. " \
                          "Install geopm service and run 'systemctl start geopm'"
```